src/main/java/betterquesting/api2/storage/LookupLogicType.java

```diff
@@ -0,0 +1,37 @@
+package betterquesting.api2.storage;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static betterquesting.api2.storage.SimpleDatabase.*;
+
+enum LookupLogicType
+{
+	ArrayCache(db -> db.mapDB.size() < CACHE_MAX_SIZE && db.mapDB.size() < SPARSE_RATIO * (db.mapDB.lastKey() - db.mapDB.firstKey()), ArrayCacheLookupLogic::new),
```

Comment:
db.mapDB.size() > SPARSE_RATIO * (db.mapDB.lastKey() - db.mapDB.firstKey())

Reply:
good catch. how could I miss this